gems/smithy/lib/smithy/templates/client/auth_plugin.erb

```diff
@@ -54,7 +54,7 @@ module <%= module_name %>
           }
 
           auth_options.each do |auth_option|
```

Comment:
Another thing to investigate - I'm open to pushing this handler code to smithy-client. We still need a generated plugin for defaults and other properties, but the handler can be made generic. It cuts down on generated code.